storage-api/src/main/java/com/connexta/transformation/storage/api/status/TransformationStatus.java

```diff
@@ -0,0 +1,119 @@
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.transformation.storage.api.status;
+
+import com.connexta.transformation.storage.api.RequestInfo;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Optional;
+
+/**
+ * An interface for classes which track the status of transformation processes. A
+ * TransformationStatus contains information about the original transformation request and the ID of
+ * the transformation the status is associated with.
+ */
+public interface TransformationStatus {
+
+  /**
+   * Returns the ID of the transformation that this status is associated with.
+   *
+   * @return the ID of the transformation that this status is associated with
+   */
+  String getTransformId();
+
+  /**
+   * Returns information about the request that started this transformation.
+   *
+   * @return information about the request that started this transformation
+   */
+  RequestInfo getRequestInfo();
+
+  /**
+   * Returns the start time of the transformation.
+   *
+   * @return the time the transformation started
+   */
+  Instant getStartTime();
+
+  /**
+   * Returns the completion time of the transformation.
+   *
+   * @return the completion time of the transformation
+   */
+  Optional<Instant> getCompletionTime();
+
+  /**
+   * Returns the duration of the transformation. If the transformation is ongoing, the difference
+   * between the start time and now will be returned, otherwise, the difference between the start
+   * and completion times will be returned.
+   *
+   * @return the duration of the transformation.
+   */
+  Duration getDuration();
+
+  /**
+   * Gets the current state of the transformation.
+   *
+   * @return the current transformation state
+   */
+  State getState();
+
+  /**
+   * Returns true if the transformation has finished, otherwise, returns false.
+   *
+   * @return true if the transformation has been completed, otherwise false
+   */
+  default boolean isCompleted() {
+    final State state = getState();
+
+    return (state == State.SUCCESSFUL || state == State.FAILED);
+  }
+
+  /**
+   * Returns true if the transformation failed, otherwise, returns false.
+   *
+   * @return true if the transformation failed, otherwise false
+   */
+  default boolean hasFailed() {
+    return (getState() == State.FAILED);
+  }
+
+  /**
+   * Checks if this transformation has successfully completed.
+   *
+   * @return true if the transformation completed successfully, otherwise false
+   */
+  default boolean wasSuccessful() {
+    return (getState() == State.SUCCESSFUL);
+  }
+
+  /** The various states a transformation can be in. */
```

Comment:
✏️ Might want to add a `isUnknown()` for completeness and this will most likely become useful when we serialize to prevent it.